lib/events/s3_exporter.rb

```diff
@@ -0,0 +1,66 @@
+module Events
+  class S3Exporter
+    def initialize(created_before, created_on_or_after = nil)
+      @created_before = created_before
+      @created_on_or_after = created_on_or_after
+    end
+
+    def export
+      file = Tempfile.new("events_s3_#{filename}.csv.gz")
+      zipped = Zlib::GzipWriter.open(file) do |gzip|
+        gzip.orig_name = "#{filename}.csv"
+        build_csv(gzip)
+        gzip.close
+      end
+      upload(zipped)
+      exported = event_scope.update_all(payload: nil)
```

Comment:
Also out of interest if we set the payload to an empty json object, then would we need the migration and monkey patch?